### ISSUE TITLE
Relative/Absolute redirection errors master list

### ISSUE THREAD

**Issue by MrPetovan** [friendica/friendica#5996]
Caused by #5907
Paging @nupplaphil
```
PHP Fatal error:  Uncaught Friendica\Network\HTTPException\InternalServerErrorException: URL is not a relative path, please use System::externalRedirectTo in src/App.php:2003
Stack trace:
#0 mod/toggle_mobile.php(20): Friendica\App->internalRedirect('https://friendi...')
#1 src/LegacyModule.php(68): toggle_mobile_init(Object(Friendica\App))
#2 src/LegacyModule.php(39): Friendica\LegacyModule::runModuleFunction('init')
#3 [internal function]: Friendica\LegacyModule::init()
#4 src/App.php(1852): call_user_func(Array)
#5 index.php(15): Friendica\App->runFrontend()
#6 {main}
  thrown in src/App.php on line 2003
```
Fixed by #6005

**Comment by MrPetovan** [comment]
```
Fatal error:  Uncaught Friendica\Network\HTTPException\InternalServerErrorException: URL is not a fully qualified URL, please use App->internalRedirect() instead in src/Core/System.php:250
Stack trace:
#0 mod/redir.php(39): Friendica\Core\System::externalRedirect('https://nerdica...')
#1 src/LegacyModule.php(68): redir_init(Object(Friendica\App))
#2 src/LegacyModule.php(39): Friendica\LegacyModule::runModuleFunction('init')
#3 [internal function]: Friendica\LegacyModule::init()
#4 src/App.php(1852): call_user_func(Array)
#5 index.php(15): Friendica\App->runFrontend()
#6 {main}
  thrown in src/Core/System.php on line 250
```

**Comment by annando** [comment]
Too much testing? :grinning:

**Comment by nupplaphil** [comment]
:-P

**Comment by MrPetovan** [comment]
#6005 only fixes the first error mentioned and adds some debug for the next time the second happens.

**Comment by nupplaphil** [comment]
Yes, because I have no idea why the externalRedirect fail with a fully qualified URL. I'm searching in the docs

**Comment by MrPetovan** [comment]
No problem, I just explained why the issue is still open after we had a PR claiming to fix it.

**Comment by nupplaphil** [comment]
So `redir` can either be used for internal and external redirects. Thanks for the report :-)

**Comment by rabuzarus** [comment]
@nupplaphil 
Yes. Friendica redirects all dfrn contacts. Or to describe it a little bit deeper. All dfrn contacts get a link to `redir/PUBLIC_CONTACT_ID`. Next to the important task to have a look if the contact is a friend and deciding about the different authentication methods, redir.php will redirect to the contact. This contact can be an external contact but also an internal contact

**Comment by MrPetovan** [comment]
Why not used absolute URLs all the time then? What's the gain of using relative URIs in this case?

**Comment by MrPetovan** [comment]
All clear for now.

**Comment by MrPetovan** [comment]
Just this morning:
```
Fatal error:  Uncaught Friendica\Network\HTTPException\InternalServerErrorException: 'https://friendica.mrpetovan.com/search?tag=King s gambit' is not a fully qualified URL, please use App->internalRedirect() instead in src/Core/System.php:251
Stack trace:
#0 mod/toggle_mobile.php(20): Friendica\Core\System::externalRedirect('https://friendi...')
#1 src/LegacyModule.php(68): toggle_mobile_init(Object(Friendica\App))
#2 src/LegacyModule.php(39): Friendica\LegacyModule::runModuleFunction('init')
#3 [internal function]: Friendica\LegacyModule::init()
#4 src/App.php(1820): call_user_func(Array)
#5 index.php(15): Friendica\App->runFrontend()
#6 {main}
  thrown in src/Core/System.php on line 251
```

**Comment by MrPetovan** [comment]
Closed by #6217
